omnipaxos_core/src/storage.rs

```diff
@@ -443,39 +460,58 @@ where
     }
 
     fn load_cache(&mut self) {
-        self.state_cache.set_decided_idx(self.storage.get_decided_idx());
-        self.state_cache.set_accepted_round(self.storage.get_accepted_round());
-        self.state_cache.set_promise(self.storage.get_promise());
-        self.state_cache.set_compacted_idx(self.storage.get_compacted_idx());
+        // try to load from storage, if storage is empty, use default values
+        match self.storage.get_promise() {
+            // set the cache to the values from storage
+            Some(promise) => {
+                self.state_cache.set_promise(promise);
+                self.state_cache.set_decided_idx(self.storage.get_decided_idx());
+                self.state_cache.set_accepted_round(self.storage.get_accepted_round().unwrap_or_default());
+                self.state_cache.set_compacted_idx(self.storage.get_compacted_idx());
+            }
+            // set the cache to default values
+            None => {
+                self.state_cache.set_promise(Ballot::default());
+                self.state_cache.set_decided_idx(0);
+                self.state_cache.set_accepted_round(Ballot::default());
+                self.state_cache.set_compacted_idx(0);
```

Comment:
Is this needed? The cache should already be initialized with default values right?